### DISCUSSION TITLE
Enhance Split View

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#813] [Ideas]
### What feature would you like to see?

Edge's Split screen has some useful features, and I hope Zen's Split View can achieve similar results.

1. Load links opened in this tab in the opposite tab.
	PS: This feature doesn't seem to work for GitHub!

	https://github.com/user-attachments/assets/e7ec444d-b3c2-4ab0-9d3d-37655d2c5930

2. Force links to load in split view instead of new tab.

	https://github.com/user-attachments/assets/b6c757fb-6565-4417-abeb-d6b4a2ea063b

3. Open the split view via the context menu.

	https://github.com/user-attachments/assets/bb3b74ca-d525-4e9b-8d73-315b7c1a3b18

4. Preview other tabs and open it in split-screen view.
	The search box on this page is similar to the browser address bar and can load URLs or search text.

	https://github.com/user-attachments/assets/17309c1f-ce4c-41ae-a89c-7f25fa104441

5. Quickly trigger common actions such as swapping tab positions, opening in a new tab, etc.

	https://github.com/user-attachments/assets/dd4d1129-6621-4e0e-8f94-9517f3538c91

### What platform are you seeing the problem on?

Windows

 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/512_

**Comment by EonZeNx** [comment]
Dragging tabs to rearrange them would be great

**Reply by IAmTahazzot** [reply]
That would be crazy awesome 🚀

**Reply by lmBored** [reply]
+1

**Comment by Iziram** [comment]
related to #892

**Comment by japostoles** [comment]
I was going to add an issue for a feature request, but found this one that has it listed already.

A +1 for "#3 - Force links to load in split view instead of new tab."

Great project, btw! :)

**Reply by fcrespo82** [reply]
It would be useful to have a setting that lets Alt+Click (configurable) open a link in a split tab, similar to how Glance works.

**Comment by JosueGalRe** [comment]
I would also suggest the ability to resize the tabs in split view mode.

Great project, currently testing it and loving it :)

**Comment by notsatvrn** [comment]
A feature I think would be nice is the option to make tab focus follow the cursor without having to click inside the tab.

**Reply by GJRobert** [reply]
This seems to be present when I start to use Zen since 2024/12/19.

**Comment by Autumnlight02** [comment]
@mauro-balades is the split view ui etc made using ts or written in a different language? Depending on that I would like to contribute a little bit.

**Reply by mr-cheffy** [reply]
Hi, it's made in JavaScript, it's under the components repository, thanks!

**Comment by KiKaraage** [comment]
I would love to see '**Search Google for (keyword)...**' opened on Split View by default (and can be turned off/on from Settings)! Would be **so much seamless and productive**.

**Comment by kittybwained** [comment]
I think dragging to resize the split and dragging splits around to reposition them would be the most useful.
Links that open in a new tab opening in a new split instead would be useful too.

**Reply by No2515** [reply]
Resizing is a must for me

**Reply by kittybwained** [reply]
Yeah, I find myself not using it because the sizes aren't quite right. Resizing is a must imo

**Reply by mr-cheffy** [reply]
We are working on this, yeah

**Reply by lmBored** [reply]
Yes, Im looking forward to that!!

**Comment by n-xlkt** [comment]
> 4. Preview other tabs and open it in split-screen view.
The search box on this page is similar to the browser address bar and can load URLs or search text.

I expected this behavior when first opening a split-view. What are the chances that the tabs you want in your grid are next to each other in your tab list? It'd make a lot more sense to provide an in-focus search box and _the option_ to select from a recent tab.

> 5. Quickly trigger common actions such as swapping tab positions, opening in a new tab, etc.

I'd like to see a double-click on a given split-view border to swap the respective adjacent views. This would be similar to double-tapping the same border on Android devices.

**Comment by rayyangh** [comment]
Hi! I love split view so far.
Edge has this split view button, I would love to see something similar on zen as an option to add to the toolbar.
![image](https://github.com/user-attachments/assets/22d02ffd-7b16-4a01-b727-1a1c2318b10b)

**Comment by MakwaWes** [comment]
I didn't see the request so adding mine here.
Have the possibility to freeze scrolling between split views.  Like if the scrolling would be mutual for the displayed split views.

PS: So much appreciation for all the devs and @mauro-balades for this awesome project. Really boosts FF to a next level !!!!

**Comment by Autumnlight02** [comment]
I also need to push my improvements. I added better split options, so you can just right click any tab and immedietly split. will do it next week / this weekend. was busy so far.

**Comment by DesBw** [comment]
Clicking a link, I also want the link to open in the other side (of the split), rather than a new unsplit tab.

**Comment by ByAldon** [comment]
Split view seem to be possible now. But how do you use it?

**Reply by GJRobert** [reply]
Most basically, in any tab, press the split view hotkey combination, and this tab and the next tab will be put in a vertically splitted view.
Other keyboard shortcuts can change to horizontal, or close the whole view.

Or, use Ctrl/Cmd/Shift-clicking on a couple of tabs in the vertical tabbar, and in the context menu select "Split n tabs" to put them.

**Reply by ByAldon** [reply]
that did not do anything what so ever...

**Reply by GJRobert** [reply]
I don't have time to record my screen now

**Comment by GJRobert** [comment]
Suggestion of a tab menu item, to assign any tab in any part of the split view
e.g. "Assign to Split View > "
- Left
- Right
- Top
- Bottom
- Top-left
- Top-right
- Bottom-left
- Bottom-right
(options depending on the then current layout of the split view)

So that, we can change which tab to view in the split view quickly :)

**Comment by IAmTahazzot** [comment]
## Indispensable features request

1. Tab splitting should be merged into a group, not like chrome group rather more like split group as all of them lives inside a single tab like ARC but more better approach maybe. Such as split tab would be shown as icons only if more than 2 or 3. Here's what I meant in the screenshot. _It is a single tab (precisely 3 tab splitting in arc)_
![Screenshot 2025-01-08 120632](https://github.com/user-attachments/assets/3d4a2a57-83ef-490b-a212-5994558da33c)

2. Reposition by dragging and custom keybinding would be great. Every major feature should have some key shortcut to work with.
3. Any **single** tab inside split group should be able to separate from split group or close easily.
4. Closing the split group at once from UI or any shortcut

**Reply by HewJhong** [reply]
+1 for point 3, that would made it much easier to work with multiple split tabs!

**Comment by DRYN07** [comment]
The Sidekick Browser had a very useful feature: ALT+Click on any link/bookmark/tab opened it on split-view. I would really love to see this in Zen.

**Reply by ByAldon** [reply]
it used to be there. But for some reason it doesnt work anymore

**Reply by KiKaraage** [reply]
Yeah would love that. Or, something that I had suggested right when Glance was developed: Add one more button to open a glanced site in split view, like in Arc

**Comment by RavioEnby** [comment]
Are we putting all Split View idea in this chat? sorry, I am not familiar with github.
I saw the developer's post in reddit that they might improve split view recently. So I would like to give some idea here.

https://github.com/zen-browser/desktop/discussions/4407#discussion-7833009

**Reply by 12th-devs** [reply]
Sure

**Comment by Sneethe** [comment]
I'd love to see simultaneous scrolling in splits. Meaning  if you were to starting scrolling on the split border (or hold down a key). scroll events would be sent to both splits. 
As soon as I saw zen browsers split feature I instantly thought of this. Namely because when comparing multiple spec sheets or other similar data, on FF the workflow would be to switch between the 2 tabs often, while remembering to do arrow down/page down on both tabs. A genuinely horrible experience.

**Reply by MakwaWes** [reply]
I agree :wink: 
https://github.com/zen-browser/desktop/discussions/813#discussioncomment-10960927

**Comment by Nanky586** [comment]
I am a power user of split views in Arc and i always wanted one simple feature that would increase speed and productivity for research immensely. 

Ability to drag and drop a link (like linked text or youtube thumbnail) from one split view window to the other to open the link in that window.

In Arc this would result in being thrown out of split view because an entirely new tab would be opened for the link.

**Reply by ByAldon** [reply]
im a power user too and like this idea. but this wont seem a easy thing to do...

**Comment by MTeaHead** [comment]
Moving windows in the split-view and changing focus between only the split windows via dedicated shortcuts would be great.   To keeping similar to the current split view keybinds, something like Ctrl+Alt+Arrow keys and Ctrl+Shift+Alt+Arrow keys could be used for changing focus and moving windows, respectively.  


Also a toggle for focus-follows-mouse would be nice.

**Reply by Nanky586** [reply]
That focus-follows-mouse would be amazing. One thing about Arc i hated was that you tend to lose track of what split-view window is in focus (especially since scroll from mouse wheel does follow the mouse) and if you use a shortcut to close or interact with the desired window, it would be the wrong one.

**Comment by cam-byte** [comment]
+1 for 1. Load links opened in this tab in the opposite tab.

Being able to configure a key to hold down to keep the Glance and be able to do like alt ctrl or something and click for split would be amazing. Super useful for if reviewing information with less button clicks. Especially if there is stuff i need to compare.

**Reply by cam-byte** [reply]
I am working on reviewing clients logos and colors for their site. I made an overview for my clients and have buttons that link to their site. At a glance works alright but covers my overview so i cant really see the colors to compare them. I have to keep opening and then closing to double check they match etc.

**Comment by pxlrbt** [comment]
It feels weird to me, that Splits open on the left of the current tab from Glance when reading order is LTR.

**Comment by bensonphillipsiv** [comment]
The "New Empty Split View" keybind should create additional split view windows when there are already two or more open.
